src/cmdline/mod.rs

```diff
@@ -256,10 +258,13 @@ impl Cmdline {
     /// # use linux_loader::cmdline::*;
     /// let mut cl = Cmdline::new(10);
     /// cl.insert_str("foobar");
-    /// assert_eq!(cl.as_str(), "foobar");
+    /// assert_eq!(cl.as_cstr(), "foobar");
     /// ```
-    pub fn as_str(&self) -> &str {
-        self.line.as_str()
+    pub fn as_cstr(&self) -> &Cstr{
```

Comment:
We should also add a regression test that validates that the string is null terminated.

Can you also add more details to the commit message? We should specify here that this was a bug introduced by #72 and that the kernel is reading from the command line until it finds the null terminator, which is why we need to be sure to add it.

Reply:
I updated the commit message. Also, fixed the tests from pipeline. I introduced various checks that the as_cstring() method returns a null terminated string. I also modified the existing unit test from loader to check for the null terminator. Should I add a separate test in linux loader just to check the null terminated string returned by as_cstring? I don't think is necessary as long as the current tests checks the bytes used for writing the cmdline into guest's memory to contain the null terminator. Let me a comment if you think there are other tests that I can add.